frontend/src/hooks/use-auto-resize.ts

```diff
@@ -189,72 +134,181 @@ export const useAutoResize = (
+    (elementHeight: number) => {
+      onHeightChange?.(elementHeight);
+      if (hasUserResizedRef.current) {
+        manualHeightRef.current = elementHeight;
+      }
+    },
+    [onHeightChange],
+  );
+
+  // Handle drag start - set manual mode flag
+  const handleDragStart = useCallback(() => {
+    hasUserResizedRef.current = true;
+    onGripDragStart?.();
+  }, [onGripDragStart]);
+
+  // Handle drag end - clear manual mode if at minimum height
+  const handleDragEnd = useCallback(() => {
+    const textareaElement = elementRef.current;
+    if (textareaElement) {
+      const currentHeight = getStyleHeightPx(textareaElement, minHeight);
+      if (Math.abs(currentHeight - minHeight) <= EPS) {
+        hasUserResizedRef.current = false;
+        manualHeightRef.current = null;
+      }
+    }
+    onGripDragEnd?.();
+  }, [minHeight, onGripDragEnd]);
 
   // Use the drag resize hook for manual resizing functionality
   const { handleGripMouseDown, handleGripTouchStart } = useDragResize({
     elementRef,
     minHeight,
     maxHeight,
-    onGripDragStart: enableManualResize ? onGripDragStart : undefined,
-    onGripDragEnd: enableManualResize ? onGripDragEnd : undefined,
-    onHeightChange,
+    onGripDragStart: enableManualResize ? handleDragStart : undefined,
+    onGripDragEnd: enableManualResize ? handleDragEnd : undefined,
+    onHeightChange: handleExternalHeightChange,
   });
 
-  // Auto-resize functionality for contenteditable div
-  const autoResize = () => {
-    const element = elementRef.current;
-    if (!element) return;
-
-    // Reset height to auto to get the actual content height
-    resetElementHeight(element);
+  // Handle content that fits within current height
+  const handleContentFitsInCurrentHeight = useCallback(
+    (element: HTMLElement, currentHeight: number, contentHeight: number) => {
+      // If user manually resized and we're above min height, preserve their chosen height
+      if (hasUserResizedRef.current && currentHeight > minHeight) {
+        applyResizeStrategy(element, {
+          finalHeight: currentHeight,
+          overflowY: "hidden",
+        });
+        executeHeightCallback(currentHeight, onHeightChange);
+        return true;
+      }
+
+      // Otherwise allow shrinking towards content (respect minHeight)
+      const finalHeight = Math.max(contentHeight, minHeight);
+      applyResizeStrategy(element, {
+        finalHeight,
+        overflowY: "hidden",
+      });
+      executeHeightCallback(finalHeight, onHeightChange);
+      return true;
+    },
+    [minHeight, onHeightChange],
+  );
 
-    // Calculate and apply optimal height
-    const optimalHeight = calculateOptimalHeight(element, constraints);
-    const finalHeight = applyHeightToElement(
-      element,
-      optimalHeight,
-      constraints,
-    );
+  // Handle content that exceeds current height but within max height
+  const handleContentExceedsCurrentHeight = useCallback(
```

Comment:
question: Why does this function return a boolean value? It appears that the return value is not being used anywhere.